### PR TITLE
Zerospeech21 vg

### LINKED ISSUE #53: W&B prompt when running an experiment
Whenever I try to run an experiment I get the following prompt:
```
wandb: (1) Create a W&B account
wandb: (2) Use an existing W&B account
wandb: (3) Don't visualize my results
wandb: Enter your choice:
```

By default, I would prefer if the user doesn't have to do anything. W&B could be inactive or use choice (3) above.

### PR DESCRIPTION
**Pull request by bhigy**
Changes related to the ZeroSpeech challenge:
 - added support for SpokenCOCO dataset
 - added code to support the use of pretrained features + utility script to extract CPC features
 - refactored tokenization helpers making the tokenizer a global variable of dataset.py
 - changed platalea default config path ~/.platalea -> ~/.config/platalea
 - disabled use of wandb by default in basic.py and transformer.py experiments
 - pinning down pytorch version

Resolves #53, #103 and #104 and (temporarily) solves #116. 

Checklist before merging:

- [x] Does this PR warrant a version bump? If so, make sure you add a git tag and GitHub release.
- [x] Did you update the changelog with a user-readable summary? Please, include references to relevant issues or PR discussions. See [CII [release_notes] criterion](https://bestpractices.coreinfrastructure.org/en/criteria/0#0.release_notes_vulns).
- [ ] When adding new functionality: did you also add a test for it?

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,18 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## [Unreleased]
```
Comment (bhigy) [inline CHANGELOG.md:9]: Should we remove this section? I don't really see a use as any change merged into the master branch should trigger a new version. Or can we have versions that don't correspond to a release?

File: requirements.txt [hunk 2]
```diff
@@ -23,7 +23,7 @@ six==1.14.0
 sortedcontainers==2.1.0
 tables==3.6.1
 tdev2 @ git+https://github.com/bootphon/tdev2.git@a72467faf209a4f4ee90e81406a07dbfebe51e00
-torch==1.2.0
+torch==1.8.1
```
Comment (bhigy) [inline requirements.txt:26]: Temporary fix for #116. Are we fine with that or shoud we solve the problem differently?

File: CHANGELOG.md (hunk 1 above)
Comment (cwmeijer) [reply CHANGELOG.md:9]: We talked about this during the meeting.

I think you should move everything under [2.0] to unreleased, right? There is no released version 2.0 yet so that header could be removed until we make the release. Alternatively we could make a release just after merging, but only if you think that makes sense as a release point.

File: platalea/basic.py [hunk 3]
```diff
@@ -84,10 +85,8 @@ def val_loss(net):
 
     if not wandb_log:
         wandb_log = config
-
-    logging.getLogger().info(
-        "Run 'wandb disabled' if you don't want to use wandb cloud logging.")
```
Comment (cwmeijer) [inline platalea/basic.py:89]: Why was this removed? I thought we decided that it would help users finding their way without having to read through the whole documentation. Did the warning go somewhere else perhaps?

File: platalea/experiments/flickr8k/basic.py [hunk 4]
```diff
@@ -53,4 +53,4 @@
                   )
 
 logging.info('Training')
-result = M.experiment(net, data, run_config)
+result = M.experiment(net, data, run_config, wandb_mode='disabled')
```
Comment (cwmeijer) [inline platalea/experiments/flickr8k/basic.py:56]: How to reenable it?
If I set 'wandb online' on the command line, won't the setting be overridden here?

File: platalea/experiments/flickr8k/transformer.py [hunk 5]
```diff
@@ -99,4 +99,5 @@ def __new__(cls, value):
 logged_config['encoder_config'].pop('SpeechEncoder')  # Object info is redundant in log.
 
 logging.info('Training')
-result = M.experiment(net, data, run_config, wandb_project='platalea_transformer', wandb_log=logged_config)
+result = M.experiment(net, data, run_config, wandb_project='platalea_transformer',
+                      wandb_log=logged_config, wandb_mode='disabled')
```
Comment (cwmeijer) [inline platalea/experiments/flickr8k/transformer.py:103]: How to reenable it?
Just like above.

**Review by cwmeijer — CHANGES_REQUESTED** [review]
Great addition 👍 
Adding a test for any of the parts would improve it though. Maybe it's better to not do that now, as this would keep the PR open for some time probably, but instead make an issue to write tests for (some of) the new parts.
I pressed 'request changes' but I'm fine with merging this when you read my comments/suggestions.

File: platalea/basic.py (hunk 3 above)
Comment (bhigy) [reply platalea/basic.py:89]: I removed it as wandb is now disabled by default in the example we provide. I just added a docstring with an explanation of `wandb_mode` parameter. Do you think that's enough or should we also provide that information elsewhere?

File: platalea/experiments/flickr8k/basic.py (hunk 4 above)
Comment (bhigy) [reply platalea/experiments/flickr8k/basic.py:56]: `wandb_mode` takes precedence over the command line and the environment variable `WANDB_MODE`. To go back to default behavior, `wandb_mode=None` should be used, which is also the default value. I clarified that in the docstring.

File: platalea/basic.py (hunk 3 above)
Comment (cwmeijer) [reply platalea/basic.py:89]: In that case it makes sense. Good change :-)

File: platalea/experiments/flickr8k/basic.py (hunk 4 above)
Comment (cwmeijer) [reply platalea/experiments/flickr8k/basic.py:56]: I can't find that docstring at the moment. Maybe it's better to have a chat about it this afternoon. Maybe we can even merge it then.
